runelite-client/src/test/java/net/runelite/client/plugins/puzzlesolver/PuzzleSolverTest.java

```diff
@@ -85,6 +86,20 @@
 
 	private static final int[] FINISHED_STATE = new int[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, -1};
 
+	@Test
+	public void testSolverMM()
```

Comment:
Test with MM puzzle states. The START_STATES are clue puzzlebox states.

Reply:
Should i test with the translated states or the raw states?

Reply:
I'd say translated states

Reply:
Some test results:

4x4 then IDA*: https://pastebin.com/xXeuiJ8u
3x4 then IDA*: https://pastebin.com/4SSiU9qK

Doing 4x4 then IDA* takes 131 steps on average but can exceed the 5 second timer on the puzzlesolver.
Doing 3x4 then IDA* takes 166 steps on average but always solves in a few 100ms.